src/librustc/middle/const_limit.rs

```diff
@@ -0,0 +1,26 @@
+
```

Comment:
I was expecting this to be part of `recursion_limit.rs`. It seems like this file is almost identical to that one? You can rename that file to `limits.rs` if that's why you separated them. Currently, `recursion_limit.rs` also handles type length limit parsing.

Reply:
I put it in a new file because I was under the impression that it will be invoked differently than the recursion_limit:
https://github.com/rust-lang/rust/blob/e9469a6aec2f49fa1e2ae670649f293866932253/src/librustc_interface/passes.rs#L207

But if this is not the case I'll rename it to `limit.rs` and put it in the same file.

As you mention below the limit is isize, wouldn't that conflict with the implementation of `recursion_limit.rs` or is there a saturating cast an option?

Reply:
I would store it as a `usize`in `Session`, then saturating cast to an `isize` in `const_eval`.

Reply:
`limit.rs` seems fine to me.